### ISSUE TITLE
Incorrect dylib load path for IntuneMAMTelemetry framework

### ISSUE THREAD

**Issue by erikkerber** [msintuneappsdk/ms-intune-app-sdk-ios#267]
**Describe the bug:**

An app built with IntuneMAMTelemetry may crash at launch due to an incorrect load path. The path appears to be what one would expect for system libraries, not a path relative to the `@rpath`

IntuneMAMSwift vs IntuneMAMTelemetry

<img width="370" alt="CleanShot 2022-08-19 at 12 34 36@2x" src="https://user-images.githubusercontent.com/75236/185677974-58d1e9db-091e-4cd8-ad29-02ae2c57df41.png">

<img width="528" alt="CleanShot 2022-08-19 at 12 34 05@2x" src="https://user-images.githubusercontent.com/75236/185677958-0f6a869e-eb76-48c4-9184-afdf218d5822.png">

**To Reproduce**
1. Link IntuneMAMTelemetry into a test bundle or app
2. Run the executable
3. Observe the crash

Xcode _may_ mask this behavior. Using an alternate build system such as Bazel will expose it.

**Expected behavior:**

Use the `@rpath` in the load commands and not crash

**Screenshots and logs:**

![image](https://user-images.githubusercontent.com/75236/185678182-7bd51591-87b9-4fe9-ad88-5f90bb41fd4a.png)

```bash
 2022-08-19 12:15:40.956 xctest[90769:73175634] (dlopen(/tmp/test_runner_work_dir.Q6VQsP/IntuneServiceTests.xctest/IntuneServiceTests,
       │  0x0109): Library not loaded: /Library/Frameworks/IntuneMAMTelemetry.framework/IntuneMAMTelemetry
7799   │   Referenced from: /private/tmp/test_runner_work_dir.Q6VQsP/IntuneServiceTests.xctest/IntuneServiceTests
7800   │   Reason: tried: '/Applications/Xcode-13.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/
       │ Contents/Resources/RuntimeRoot/Library/Frameworks/IntuneMAMTelemetry.framework/IntuneMAMTelemetry' (no such file), '/Library/Frameworks/IntuneMAMTelemetry.framew
       │ ork/IntuneMAMTelemetry' (no such file), '/Applications/Xcode-13.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/R
       │ untimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/IntuneMAMTelemetry.framework/IntuneMAMTelemetry' (no such file))
```

**Smartphone (please complete the following information):**
N/A

**Intune App SDK for iOS (please complete the following information):**

All 6.0.0+

**Comment by erikkerber** [comment]
This can be worked around by doctoring the supplied xcframework/framework

```
install_name_tool -id @rpath/IntuneMAMTelemetry.framework/IntuneMAMTelemetry IntuneMAMTelemetry.xcframework/ios-arm64_x86_64-simulator/IntuneMAMTelemetry.framework/IntuneMAMTelemetry
install_name_tool -id @rpath/IntuneMAMTelemetry.framework/IntuneMAMTelemetry IntuneMAMTelemetry.xcframework/ios-arm64/IntuneMAMTelemetry.framework/IntuneMAMTelemetry
```

**Comment by neiljac1** [comment]
Hi @erikkerber . Thanks for reporting this issue. Another work around would be to simply include IntuneMAMTelemetry.framework in your application's Frameworks directory. The Intune MAM SDK will load it dynamically. Your application doesn't need to link to it directly.

Thanks,
Neil

**Comment by erikkerber** [comment]
Hey @neiljac1! It is already included in Frameworks, but since the load path is pointing to the system library directories it can't be found. 

It's a bit of a bulge of text, but the locations the runtime looks for it are listed in the last snippet above:

```
Reason: tried: '/Applications/Xcode-13.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/
       │ Contents/Resources/RuntimeRoot/Library/Frameworks/IntuneMAMTelemetry.framework/IntuneMAMTelemetry' (no such file), '/Library/Frameworks/IntuneMAMTelemetry.framew
       │ ork/IntuneMAMTelemetry' (no such file), '/Applications/Xcode-13.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/R
       │ untimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/IntuneMAMTelemetry.framework/IntuneMAMTelemetry' (no such file))
```

**Comment by erikkerber** [comment]
I am able to work around the issue for now by patching the SDK in-situ from our builds using a variation of the command above. It works for tests, and I assume (but we'll know soon) that it fixes on device as well.

**Comment by erikkerber** [comment]
@neiljac1 was this fixed intentionally? I don't appear to see it in 18.0.1, but also haven't checked recent versions.

**Comment by neiljac1** [comment]
Hi @erikkerber, yes this was fixed in 17.7.2.

Thanks,
Neil
